package dao;

import java.sql.Connection;
import java.sql.Date;
import java.sql.PreparedStatement;
import java.sql.ResultSet;

import beans.Endereco;
import beans.Usuario;
import connection.Conexao;

public class UsuarioDAO {
	private Connection con;
	private PreparedStatement stmt;
	private ResultSet rs;
	
	public UsuarioDAO() throws Exception {
		con = new Conexao().conectar();
	}
	
	/**
	 * Responsável por adicionar uma linha na tabela T_SIC_USUARIO
	 * @param u Recebe um objeto do Tipo Usuario Beans
	 * @return Uma string com a mensagem de confirmação.
	 * @throws Exception Chamada da exceção checked.
	 * @author Fábio Salgarella
	 */
	
	public String gravar(Usuario u) throws Exception{
		PreparedStatement stmt = con.prepareStatement("INSERT INTO T_SIC_USUARIO (CD_CLIENTE, NM_CLIENTE, DS_EMAIL, DS_SENHA, DS_SEXO, DT_NASCIMENTO, NR_CPF, DS_ENDERECO, NR_TIPO  ) VALUES (?,?,?,?,?,?,?,?,?)");
		
		stmt.setInt(1, u.getCodigo());
		stmt.setString(2, u.getNome());
		stmt.setString(3, u.getEmail());
		stmt.setString(4, u.getSenha());
		stmt.setString(5, u.getSexo());
		stmt.setDate(6, (Date) u.getNascimento());
		stmt.setString(7, u.getCpf());
		stmt.setInt(8, u.getEndereco().getCodigo());
		stmt.setInt(9, u.getTipo());
		
		stmt.executeUpdate();
		
		return "Cadastrado com Sucesso!";
	}
}
